### PR TITLE
Fixed "Unable to find source stat item" error

### PR DESCRIPTION
**Pull request by johnc303**
Added fix for sourceKey being incorrect, which then caused the source stat to not be found. Seemed to be caused by object pointer issues so deep cloned items before creation.

### CODE REVIEW

File: module/actor/actor.js [hunk 1]
```diff
@@ -509,13 +509,16 @@ export class PrimePCActor extends Actor
...
 			{
 				if (item.type == statType && item.system.default)
 				{
-					item.system.sourceKey = item.id;
-					actorItemsToCreate.push(item);
-					statItem = this._getItemDataAsStat(item);
+					// Deep clone it to prevent object point related weirdness
+					const itemClone = JSON.parse(JSON.stringify(item));
+					console.log(`Updating sourceKey. Old: '${itemClone.system.sourceKey}', New:'${itemClone._id}'`);
+					itemClone.system.sourceKey = itemClone._id;
```
Comment (biz812) [inline module/actor/actor.js:519]: After we set the clone's sourceKey to the ID of the Item we clone it from, shouldn't we then give it a new, unique ID of it's own?
e.g. `itemClone._id = <generated_id>`

Comment (biz812) [reply module/actor/actor.js:519]: Perhaps it doesn't matter because it will become an embedded item 🤔

Comment (johnc303) [reply module/actor/actor.js:519]: Good spot, but no! If you get a "document", or item in this case, and pass it to the actor's `createEmbeddedDocuments()` method, Foundry will create a clone with a new ID and a parent property set to the actor.

Comment (biz812) [reply module/actor/actor.js:519]: I did try and discern that from the API docs but they weren't much use.

**Comment by biz812** [comment]
Ship it! 🚢

**Pull request merged**